src/processor/reconnection-graph.service.ts

```diff
@@ -108,10 +117,24 @@ export class ReconnectionGraphService implements OnApplicationBootstrap, OnAppli
 
     let payWithCapacityBatchAllOp = ExtrinsicHelper.payWithCapacityBatchAll(providerKeys, calls);
     const [batchCompletedEvent, eventMap] = await payWithCapacityBatchAllOp.signAndSend();
+    if (batchCompletedEvent &&
+      !(ExtrinsicHelper.api.events.utility.BatchCompleted.is(batchCompletedEvent))) {
+        this.graphUpdateQueue.add('graphUpdate', data_nt, { jobId: jobId_nt });
+      return;
+    }
 
-    // TODO
-    // Re-import DSNP Graph from chain & verify
-    //     (if updating connections as well, do the same for connections--but do not transitively update connections - of - connections)
+    // On successful export to chain, re-import the user's DSNP Graph from the blockchain and form import bundles
+    // import bundles are used to import the user's DSNP Graph into the graph SDK
+    // check if user graph exists in the graph SDK else queue a graph update job
+    const reImported = await this.importBundles(dsnpUserId, providerId, graphSdkConfig, graphKeyPairs, graphConnections, updateConnections);
+    if (reImported) {
+      const userGraphExists = await this.graphStateManager.graphContainsUser(dsnpUserId.toString());
+      if (!userGraphExists) {
+        this.graphUpdateQueue.add('graphUpdate', data_nt, { jobId: jobId_nt });
+      }
+    } else {
+      this.graphUpdateQueue.add('graphUpdate', data_nt, { jobId: jobId_nt });
+    }
```

Comment:
reimport the graph set it in state and check if user graph is set, if not do a non-transitive update